gapic/src/main/com/google/gapid/perfetto/views/RootPanel.java

```diff
@@ -184,39 +177,7 @@ public void render(RenderContext ctx, Repainter repainter) {
     }
   }
 
-  private void renderVSync(RenderContext ctx, Repainter repainter, VSync vsync) {
-    ctx.trace("VSync", () -> {
-      VSync.Data data = vsync.getData(state.toRequest(), (future, consumer) -> {
-        state.thenOnUiThread(future, result -> {
-          consumer.accept(result);
-          repainter.repaint(new Area(0, 0, width, height));
-        });
-      });
-      if (data == null) {
-        return;
-      }
-
-      TimeSpan visible = state.getVisibleTime();
-      ctx.setBackgroundColor(colors().vsyncBackground);
-      boolean fill = !data.fillFirst;
-      double lastX = LABEL_WIDTH;
-      double h = bottom.getPreferredHeight();
-      for (long time : data.ts) {
-        fill = !fill;
-        if (time < visible.start) {
-          continue;
-        }
-        double x = LABEL_WIDTH + state.timeToPx(time);
-        if (fill) {
-          ctx.fillRect(lastX, 0, x - lastX, h);
-        }
-        lastX = x;
-        if (time > visible.end) {
-          break;
-        }
-      }
-    });
-  }
+  protected abstract void preMainUiRender(RenderContext ctx, Repainter repainter);
```

Comment:
nit: Maybe putting abstract methods close to each other, so that it's easier for future people to browse the code.